gpkit/constraints/set.py

```diff
@@ -44,6 +44,7 @@ def __init__(self, constraints, substitutions=None):
                         # so we can catch them (see above) in ConstraintSets
             if hasattr(self[i], "substitutions"):
                 self.substitutions.update(self[i].substitutions)
+                self[i].substitutions = self.substitutions
```

Comment:
1) self[i] might be the ConstraintSetted version of constraint

2) I just changed it so that the substitutions _would_ get mutated precisely because that was the behaviour people in learn3 expected!